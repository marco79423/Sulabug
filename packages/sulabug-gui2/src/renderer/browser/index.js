import React, {useEffect} from 'react'
import {useDispatch} from 'react-redux'
import {Redirect, Route, Switch} from 'react-router-dom'
import {createMuiTheme, ThemeProvider} from '@material-ui/core/styles'
import {lime, teal} from '@material-ui/core/colors'

import CollectionPage from './pages/CollectionPage'
import BrowsePage from './pages/BrowsePage'
import SettingsPage from './pages/SettingsPage'
import DownloadPage from './pages/DownloadPage'


export default function Browser() {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch({type: 'common/initialize-app', data: 'browser'})
  }, [])

  const theme = createMuiTheme({
    typography: {
      h2: {
        fontSize: '1.6rem',
      },
      subtitle2: {
        fontSize: '0.8rem',
      },
      body2: {
        fontSize: '0.9rem',
      }
    },
    palette: {
      primary: {
        main: teal[800]
      },
      secondary: {
        main: lime[600]
      },
    },
  })

  return (
    <ThemeProvider theme={theme}>
      <Switch>
        <Redirect exact from='/browser' to='/browser/collection'/>
        <Route exact path='/browser/collection' component={CollectionPage}/>
        <Route exact path='/browser/browse' component={BrowsePage}/>
        <Route exact path='/browser/download' component={DownloadPage}/>
        <Route exact path='/browser/settings' component={SettingsPage}/>
      </Switch>
    </ThemeProvider>
  )
}
