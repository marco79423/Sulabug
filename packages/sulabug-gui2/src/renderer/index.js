import React from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter as Router, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'
import {Provider} from 'react-redux'

import configureStore from './configureStore'
import Browser from './browser'

const store = configureStore()

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Redirect exact from='/' to='/browser'/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Router>
    </Provider>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
