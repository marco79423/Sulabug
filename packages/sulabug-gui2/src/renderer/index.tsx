import React, {lazy, Suspense} from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'

const Manager = lazy(() => import('./manager'))
const Browser = lazy(() => import('./browser'))

function App() {
  return (
    <HashRouter>
      <Suspense fallback={<div>loading</div>}>
        <Switch>
          <Redirect exact from='/' to='/manager'/>
          <Route path="/manager" component={Manager}/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Suspense>
    </HashRouter>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
