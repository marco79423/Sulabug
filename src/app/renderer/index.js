import * as React from 'react'
import ReactDOM from 'react-dom'

import 'reflect-metadata'

import App from '../../imports/app/App'

ReactDOM.render(
  <App/>,
  document.getElementById('app')
)
