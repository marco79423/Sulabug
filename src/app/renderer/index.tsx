import * as React from 'react'
import {render} from 'react-dom'
import App from '../../imports/ui/pages/App'

render(
  <App/>,
  document.getElementById('app')
)