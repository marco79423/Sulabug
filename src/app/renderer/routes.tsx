import React from 'react'
import { Router, Route, Switch } from 'react-router'
import createBrowserHistory from 'history/createBrowserHistory'

import BrowsePage from '../../imports/ui/pages/App'

const browserHistory = createBrowserHistory()

export const renderRoutes = () => (
  <Router history={browserHistory}>
    <Switch>
      <Route exact path="/" component={BrowsePage}/>
    </Switch>
  </Router>
)
