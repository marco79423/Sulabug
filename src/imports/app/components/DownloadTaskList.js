import * as React from 'react'
import {createStyles, withStyles} from '@material-ui/core/styles'
import List from '@material-ui/core/List'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

import DownloadTaskListItem from './DownloadTaskListItem'
import LinearProgress from '@material-ui/core/LinearProgress/LinearProgress'

const styles = (theme) => createStyles({
  card: {
    margin: theme.spacing.unit * 2,
  },
  root: {
    padding: 0,
  },
  progress: {
    marginRight: theme.spacing.unit * 2
  }
})

export class DownloadTaskList extends React.Component {
  render() {
    const {classes, loading, downloadTasks} = this.props
    if (loading) {
      return <LinearProgress color="secondary" variant="query"/>
    }

    if (downloadTasks.length === 0) {
      return (
        <Card className={classes.card}>
          <CardContent>
            <Typography variant="h2" gutterBottom>報告老大！</Typography>
            <Typography variant="body1">沒有正在下載的任務！</Typography>
          </CardContent>
        </Card>
      )
    }

    return (
      <div className={classes.root}>
        <List>
          {downloadTasks.map((downloadTask) => <DownloadTaskListItem key={downloadTask.id}
                                                                     downloadTask={downloadTask}
                                                                     deleteDownloadTask={this.props.deleteDownloadTask}/>)}
        </List>
      </div>
    )
  }
}

export default withStyles(styles)(DownloadTaskList)
