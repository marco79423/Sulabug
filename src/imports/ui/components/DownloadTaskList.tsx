import * as React from 'react'
import {createStyles, Theme, withStyles} from '@material-ui/core/styles'
import List from '@material-ui/core/List'

import DownloadTaskListItem from './DownloadTaskListItem'

const styles = (theme: Theme) => createStyles({
  root: {
    width: '100%',
    maxWidth: 400,
    backgroundColor: theme.palette.background.paper,
  },
  progress: {
    marginRight: theme.spacing.unit * 2
  }
})

interface PropsTypes {
  classes: any,
  downloadTasks: any,
}

export class DownloadTaskList extends React.Component<PropsTypes, {}> {
  render() {
    const {classes, downloadTasks} = this.props
    return (
      <div className={classes.root}>
        <List>
          {downloadTasks.map((downloadTask: any) => <DownloadTaskListItem key={downloadTask.id} downloadTask={downloadTask}/>)}
        </List>
      </div>
    )
  }
}

export default withStyles(styles)(DownloadTaskList)
