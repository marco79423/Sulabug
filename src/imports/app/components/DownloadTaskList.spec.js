import 'reflect-metadata'

import React from 'react'
import renderer from 'react-test-renderer'

import DownloadTaskList from './DownloadTaskList'


describe('DownloadTaskList', () => {
  it('will render download task list correctly', async () => {
    const component = renderer.create(
      <DownloadTaskList loading={false}
                        downloadTasks={[]}
                        deleteDownloadTask={() => {
                        }}/>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('will render download task list correctly in loading state', async () => {
    const component = renderer.create(
      <DownloadTaskList loading={true}
                        downloadTasks={[]}
                        deleteDownloadTask={() => {
                        }}/>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
