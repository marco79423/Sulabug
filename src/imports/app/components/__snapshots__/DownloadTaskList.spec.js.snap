// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadTaskList will render download task list correctly 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation1-8 MuiPaper-rounded-6 MuiCard-root-4 DownloadTaskList-card-1"
>
  <div
    className="MuiCardContent-root-32"
  >
    <h2
      className="MuiTypography-root-33 MuiTypography-h2-46 MuiTypography-gutterBottom-60"
    >
      報告老大！
    </h2>
    <p
      className="MuiTypography-root-33 MuiTypography-body1-42"
    >
      沒有正在下載的任務！
    </p>
  </div>
</div>
`;

exports[`DownloadTaskList will render download task list correctly in loading state 1`] = `
<div
  className="MuiLinearProgress-root-69 MuiLinearProgress-colorSecondary-71 MuiLinearProgress-query-75"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar-79 MuiLinearProgress-barColorSecondary-81 MuiLinearProgress-bar1Indeterminate-82"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-79 MuiLinearProgress-barColorSecondary-81 MuiLinearProgress-bar2Indeterminate-85"
    style={Object {}}
  />
</div>
`;
